docs/developer/development/ci.rst

```diff
@@ -0,0 +1,147 @@
+Continuous Testing
+==================
+
+Add a new test in the continuous integration system
+---------------------------------------------------
+
+A test as continuous integration system level is an End-to-End test that need a
+dedicated environment (with one or several machines (virtual or container)).
+
+A test that only check a specific feature of a classic MetalK8s deployment
+should be part of PyTest BDD and not integrated as a dedicated stage in
+continuous integration system.
+
+How to choose between Pre-merge and Post-merge
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Choice really depend on the goal of this test.
+
+As high level view:
+
+Pre-merge:
+
+- Test is not too long (less than 30 minutes).
+- Test essential feature of the product (installation, expansion, building,
+  ...).
+
+Post-merge:
+
+- Test last longer (more than 30 minutes).
+- Test "non-essential" feature of the product (upgrade, downgrade, solutions,
+  ...).
+
+How to add a stage in continuous integration system
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Continuous integration system is controled by the ``eve/main.yml`` YAML file.
+
+A stage is defined by a worker and a list of steps. Each stage should be in
+``stages`` section and trigger by ``pre-merge`` or ``post-merge``.
+
+To know the different kind of worker available, all the builtin steps, how to
+trigger a stage, ... please refer to the eve documentation.
+
+A test stage in MetalK8s context
+""""""""""""""""""""""""""""""""
+
+In MetalK8s context each test stage (eve stage that represent a full test)
+should generate a status file containing the result of the test, either a
+success or a failure, and a JUnit file containing the result of the test
+and information about this test.
+
+To generate these file each stage need some information to generate this
```

Comment:
Each stage needs the following information to generate the JUnit file: or

To generate the JUnite file, each stage needs the following information